src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectTestingTestSource.stg

```diff
@@ -0,0 +1,1101 @@
+    //CollectionTypeFlag collection_flag;// Unused. No flags apply
+    
+    $relation_capname$ArrayHeader $array.name$_$relation_name$_header = $array.name$_$relation_name$_array.header();
+    $if(relation)$
+    CompleteTypeDetail $array.name$_$relation_name$_detail = $array.name$_$relation_name$_header.detail();
+    $check_optional_detail_annotations(ctx=ctx, object=array)$
+    QualifiedTypeName $array.name$_$relation_name$_detail_type_name = $array.name$_$relation_name$_detail.type_name();
+    ASSERT_EQ("$array.scopedname$", $array.name$_$relation_name$_detail_type_name);
+    $endif$
+
+    CommonArrayHeader $array.name$_$relation_name$_header_common = $array.name$_$relation_name$_header.common();
+    LBoundSeq $array.name$_$relation_name$_header_common_bound_seq = $array.name$_$relation_name$_header_common.bound_seq();
+
+    int dimension_bound_index = 0;
+    $array.typecode.dimensions:{ dimension_bound |
+    ASSERT_EQ($dimension_bound$, $array.name$_$relation_name$_header_common_bound_seq.at(dimension_bound_index));
+    dimension_bound_index++;
+    };separator="\n"$
+
+    $relation_capname$CollectionElement $array.name$_$relation_name$_collection_element = $array.name$_$relation_name$_array.element();
+    CommonCollectionElement $array.name$_$relation_name$_common_element = $array.name$_$relation_name$_collection_element.common();
+
+    CollectionElementFlag $array.name$_$relation_name$_common_element_flags = $array.name$_$relation_name$_common_element.element_flags();
+    {
+    $check_member_flags(ctx=ctx, object=array, flags_var_name=["array"])$
+    ASSERT_TRUE(array_expected_flags & $map.name$_$relation_name$_array);
+    }
+    $get_type_identifier(ctx=ctx, type=array, relation=relation, relation_name=relation_name, var_name=["common_element_id"])$
+    ASSERT_EQ($array.name$_$relation_name$_common_element_id->_d(), $array.name$_$relation_name$_common_element.type()._d());
+
+    $if(relation)$
+{
+    CompleteElementDetail $array.name$_$relation_name$_detail = $array.name$_$relation_name$_collection_element.detail();
+    $check_optional_detail_annotations(ctx=ctx, object=array)$
+}
+    $endif$
+
+    int $array.name$_$relation_name$_totalsize = $member.typecode.dimensions:{ dim | $dim$};separator=" + "$;
+    
+    if($array.name$_$relation_name$_totalsize <= 100){
+        SBoundSeq $array.name$_$relation_name$_s_bound_seq;
+        $member.typecode.dimensions:{ dim | $array.name$_$relation_name$_s_bound_seq.push_back($dim$);};separator="\n"$
+        ASSERT_EQ($array.name$_$relation_name$_s_bound_seq, $array.name$_$relation_name$_type_identifier.array_sdefn().array_bound_seq());
+    }else{
+        LBoundSeq $array.name$_$relation_name$_l_bound_seq;
+        $member.typecode.dimensions:{ dim | $array.name$_$relation_name$_l_bound_seq.push_back($dim$);};separator="\n"$
+        ASSERT_EQ($array.name$_$relation_name$_l_bound_seq, $array.name$_$relation_name$_type_identifier.array_ldefn().array_bound_seq());
+    }
```

Comment:
This is not correct. It does not depend on the sum of the elements in each dimension, but on the number of elements of a specific dimension being greater than 255.